### PR TITLE
Fix #2608: Brave Ads notifications disappear without user input

### PR DESCRIPTION
**Pull request by tmancey**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by tmancey** [comment]
This PR fixes https://github.com/brave/brave-browser/issues/2608

File: build/mac/tweak_info_plist.py [hunk 1]
```diff
@@ -98,6 +98,9 @@ def Main(argv):
   # Explicitly disable profiling
   plist['SUEnableSystemProfiling'] = False
 
+  # Explicitly force user notification style to use alerts
+  plist['NSUserNotificationAlertStyle'] = "alert"
```
Comment (bridiver) [inline build/mac/tweak_info_plist.py]: There are other implications for this with web notifications. I think this needs more discussion/review as it changes behavior for the entire app

Comment (bridiver) [reply build/mac/tweak_info_plist.py]: @tmancey have you tested web notifications with this change? https://www.bennish.net/web-notifications.html
Also what happens if the user then changes the setting to banner? Do any notifications appear?

Comment (tmancey) [reply build/mac/tweak_info_plist.py]: Tested with web notifications which also do not dismiss until manually closed. With regards to changing the setting from Alerts to Banner in macOS/Preferences/Notifications then web notifications work as they did before the change

Comment (mrose17) [reply build/mac/tweak_info_plist.py]: If the user changes the setting to banner, they'll get the less reliable behavior.  Or the could turn it off.  Either is fine.  We just want it to work best in the default case.

Comment (tmancey) [reply build/mac/tweak_info_plist.py]: Requested change was reverted

**Pull request merged**